libc-bottom-half/sources/__wasilibc_initialize_environ.c

```diff
@@ -11,7 +11,7 @@
 /// Statically-initialize it to an invalid pointer value so that we can
 /// detect if it's been explicitly initialized (we can't use `NULL` because
 /// `clearenv` sets it to NULL.
-char **__wasilibc_environ __attribute__((weak)) = (char **)-1;
+char **__wasilibc_environ weak = (char **)-1;
```

Comment:
I think this reads way easier with `weak` at the start of the line.. and I think that seems to be how musl uses it.